### PR TITLE
Upgrade tutorial to Django 3.2

### LINKED ISSUE #1677: Update Django Girls tutorial to Django 3.1
Currently Django released 3.1.3 version. Due to this new people who follow it get in to lots of trouble. I found this from answering then in gitter channel of Djangogirls.

Is there any plans of publishing updated version of tutorial?

At least updating to version 3.0 can resolve most issues.

### PR DESCRIPTION
**Pull request by adamantike**
This diff includes the needed changes for the tutorial to be available for Django 3.2 in the English version.

Most of these changes were already available in the `django-3.1` branch, and the #1678 pull request. Big props to @nikhiljohn10 for the huge initial effort!

Closes #1677

### CODE REVIEW

File: en/deploy/README.md [hunk 1]
```diff
@@ -245,7 +245,7 @@ You can also go to the "Files" page and navigate around using PythonAnywhere's b
 Your site should now be live on the public Internet!  Click through to the PythonAnywhere "Web" page to get a link to it. You can share this with anyone you want. :)
 
 
-> **Note** This is a beginners' tutorial, and in deploying this site we've taken a few shortcuts which aren't ideal from a security point of view.  If and when you decide to build on this project, or start a new project, you should review the [Django deployment checklist](https://docs.djangoproject.com/en/2.2/howto/deployment/checklist/) for some tips on securing your site.
+> **Note** This is a beginners' tutorial, and in deploying this site we've taken a few shortcuts which aren't ideal from a security point of view.  If and when you decide to build on this project, or start a new project, you should review the [Django deployment checklist](https://docs.djangoproject.com/en/3.2/howto/deployment/checklist/) for some tips on securing your site.
```
Comment (ekohl) [inline en/deploy/README.md:248]: Should we introduce a variable like `django_minor`, `django_doc_version` or something like that?

File: en/django_urls/README.md [hunk 2]
```diff
@@ -93,10 +93,23 @@ The last part, `name='post_list'`, is the name of the URL that will be used to i
 
 If you try to visit http://127.0.0.1:8000/ now, then you'll find some sort of 'web page not available' message. This is because the server (remember typing `runserver`?) is no longer running. Take a look at your server console window to find out why.
 
-![Error](images/error1.png)
+{% filename %}{{ warning_icon }} command-line{% endfilename %}
+
+```shell
```
Comment (ekohl) [inline en/django_urls/README.md:98]: I'm not sure this really is shell.

File: en/django_urls/README.md [hunk 3]
```diff
@@ -93,10 +93,23 @@ The last part, `name='post_list'`, is the name of the URL that will be used to i
...
+    return _bootstrap._gcd_import(name[level:], package, level)
+  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
+  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
+  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
+  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
+  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
+  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
+  File "/Users/dana/Codes/djangogirls/blog/urls.py", line 5, in <module>
```
Comment (ekohl) [inline en/django_urls/README.md:106]: Typically the name `ola` is used
```suggestion
  File "/Users/ola/Codes/djangogirls/blog/urls.py", line 5, in <module>
```

File: en/extend_your_application/README.md [hunk 4]
```diff
@@ -75,7 +75,20 @@ That means if you enter `http://127.0.0.1:8000/post/5/` into your browser, Djang
...
+
+```shell
+    return _bootstrap._gcd_import(name[level:], package, level)
+  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
+  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
+  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
+  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
+  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
+  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
+  File "/Users/dana/Codes/djangogirls/blog/urls.py", line 6, in <module>
```
Comment (ekohl) [inline en/extend_your_application/README.md:88]:
```suggestion
  File "/Users/ola/Codes/djangogirls/blog/urls.py", line 6, in <module>
```

File: en/deploy/README.md (hunk 1 above)
Comment (adamantike) [reply en/deploy/README.md:248]: I think we could, though is there any gotchas we should consider for this templating system where the variable could not be replaced? (e.g. related to #1694?)

Maybe this is something we can try to tackle in a separate PR. Django docs are also available in other languages, so playing around to replace the `/en/` in the docs URL could be useful too.

File: en/django_urls/README.md (hunk 2 above)
Comment (adamantike) [reply en/django_urls/README.md:98]: It's actually stdout, as it's generated by the `runserver` command, but I don't think we will get any syntax highlighting, no matter the language we specify here. Are we OK with it just being ```?

File: en/django_urls/README.md (hunk 3 above)
Comment (adamantike) [reply en/django_urls/README.md:106]: What about the rest of the path? Do you have any convention? Preferences about it prioritizing Windows/OSX/Linux?

File: en/django_urls/README.md (hunk 2 above)
Comment (ekohl) [reply en/django_urls/README.md:98]: Yes, I think that's fine.

File: en/django_urls/README.md (hunk 3 above)
Comment (ekohl) [reply en/django_urls/README.md:106]: If you look at https://tutorial.djangogirls.org/en/django_start_project/ it's in `~/djangogirls`. I'm not sure about which OS to prioritize.

Comment (adamantike) [reply en/django_urls/README.md:106]: Noted, let's go with `/Users/ola/djangogirls/...` for now!

File: en/deploy/README.md (hunk 1 above)
Comment (ekohl) [reply en/deploy/README.md:248]: Sounds ok to me.

**Review by ekohl — COMMENTED** [review]
I didn't follow the whole tutorial, but this looks ok to me.

**Review by das-g — CHANGES_REQUESTED** [review]
There's another Django 2.2 docs URL [in `en/dynamic_data_in_templates/README.md`](https://github.com/DjangoGirls/tutorial/blob/1a47238ae20fb7ceab6109dbf372257f8bf35a6b/en/dynamic_data_in_templates/README.md?plain=1#L78).

**Comment by adamantike** [comment]
Good catch! Fixed the pending link, and double-checked again. Feel free to review one more time.

**Comment by das-g** [comment]
Has anyone gone through and tested the whole tutorial with these changes? (I didn't have time to do so, yet, and might not get to it anytime soon.)

If not, shall we merge this anyway already, and just wait whether readers report any discrepancies? I'd much prefer if this small but major change was properly tested before releasing, but I also don't want to drag it out too long.

**Comment by ekohl** [comment]
I'm in the same boat @das-g so I'm also leaning to merging.

**Review by ekohl — APPROVED** [review]

**Comment by adamantike** [comment]
To try to automate these kinds of checks in the short term, considering there's no much time available, would a simple Dockerfile that runs these commands and apply incremental `diff` files to the Django project be useful?

At least, that will allow building/running the container with a specific Python version, check the stdout to compare with the provided ones in the tutorial, and review that the Django server correctly runs at the end of the tutorial. It would involve duplicating most of the code and the steps provided in the tutorial, though.

**Comment by das-g** [comment]
> To try to automate these kinds of checks in the short term, considering there's no much time available, would a simple Dockerfile that runs these commands and apply incremental `diff` files to the Django project be useful?
> 
> At least, that will allow building/running the container with a specific Python version, check the stdout to compare with the provided ones in the tutorial, and review that the Django server correctly runs at the end of the tutorial. It would involve duplicating most of the code and the steps provided in the tutorial, though.

Setting that up would probably require about as much time as going through the tutorial manually. So it'd only pay off, it can be re-used for checking further changes. That would mean that the duplicated code would have to be kept around and to be kept in sync with the code in the tutorial, which already sounds like quite an anti-pattern.

If we wanted something like that, there should be a single source of truth from which both the rendered tutorial as well as the code to be tested can be derived completely automatically. That'd probably require using something like [Python doctest](https://docs.python.org/3/library/doctest.html) or similar.

While that be pretty cool, it's quite different from our current setup, and I don't know whether the hassle to migrate to a different workflow (that we'd have to define / figure out first) would be worth the hassle.

**Comment by ekohl** [comment]
Given there's been no response in 2 weeks, I'm leaning to merging. Right now @das-g has requested changes, which I think are addressed but that's why I won't push the big green button now.

**Review by das-g — APPROVED** [review]

**Comment by das-g** [comment]
Sorry I was (and will be in the near future) busy with other stuff.

:shipit: Let's merge and see if anyone complains about something being broken. :see_no_evil:

**Comment by adamantike** [comment]
Thank you for coming back to it!

If any issue is reported regarding the upgrade, I'm available to work on it.

**Pull request merged**